### PR TITLE
Extract export database logic into own class

### PR DESCRIPTION
**Pull request by XiangRongLin**
#### What is it?
- [x] Codebase improvement (dev facing)

#### Description of the changes in your PR

- Separate database/settings generation logic from the UI.
- Add happy path unit test.

With this seperation, unit tests can be introduced. I just added one as proof of concept for now.
If this style is ok i would work on extracting more stuff out.

This also allows cheap unit tests instead of comparatively expensive androidTests.

#### APK testing 

[app-debug.zip](https://github.com/TeamNewPipe/NewPipe/files/5624885/app-debug.zip)

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by XiangRongLin** [comment]
I added the setting `Mockito.withSettings().stubOnly()` in the test, since a stub is enough

**Review by Stypox — APPROVED** [review]
We definitely need more tests, so I like this. I tested on my device and it works; I also run the test locally and it works. Thank you!

File: app/src/test/java/org/schabi/newpipe/settings/ContentSettingsManagerTest.kt [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+class ContentSettingsManagerTest {
+
+    @RunWith(MockitoJUnitRunner::class)
+    class ExportTest {
+
+        private lateinit var preferences: SharedPreferences
+        private lateinit var newpipeDb: File
+        private lateinit var newpipeSettings: File
+
+        @Before
```
Comment (XiangRongLin) [inline app/src/test/java/org/schabi/newpipe/settings/ContentSettingsManagerTest.kt:29]: Well from refactoring too much stuff around, i apparently created a `beforeClass` method annotated `@Before`. I'll fix this later on in my next PR.

Comment (TobiGr) [reply app/src/test/java/org/schabi/newpipe/settings/ContentSettingsManagerTest.kt:29]: Please create a separate PR for this

File: app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java [hunk 2]
```diff
@@ -131,6 +129,8 @@ public void onCreatePreferences(final Bundle savedInstanceState, final String ro
         newpipeSettings = new File(homeDir + "/databases/newpipe.settings");
         newpipeSettings.delete();
 
+        manager = new ContentSettingsManager(homeDir);
```
Comment (TobiGr) [inline app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java:132]: That causes the build to fail. Can you fix that, please.

Comment (Stypox) [reply app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java:132]: Oh, how is this possible? I ran the tests successfully on my machine...

Comment (XiangRongLin) [reply app/src/main/java/org/schabi/newpipe/settings/ContentSettingsFragment.java:132]: WOW, i see where this is coming from. PR coming

**Pull request merged**